core/src/main/java/org/apache/stormcrawler/pii/PiiBolt.java

```diff
@@ -0,0 +1,194 @@
+	
+	public static final byte[] REDACTED_BYTES = REDACTED_HTML.getBytes(StandardCharsets.UTF_8);
+
+	/**
+	 * Returns a Scheduler instance based on the configuration *
+	 */
+	public static PiiRedacter getInstance(Map<String, Object> stormConf) {
+		PiiRedacter redacter;
+
+		String className = ConfUtils.getString(stormConf, PII_REDACTER_CLASS_PARAM);
+		if (className == null || className.isEmpty()) {
+			throw new RuntimeException("PiiRedacter class name must be defined in the configuration (pii.redacter.class)");
+		}
+
+		LOG.info("Loading PII Redacter class, name={}", className);
+		try {
+			redacter = InitialisationUtil.initializeFromQualifiedName(className, PiiRedacter.class);
+		} catch (Exception e) {
+			throw new RuntimeException("Can't instantiate " + className, e);
+		}
+
+		LOG.info("Initializing PII Redacter instance");
+		try {
+			redacter.init(stormConf);
+		} catch (Exception e) {
+			LOG.error("Error while initializing PII Redacter", e);
+		}
+
+		return redacter;
+	}
+
+	public void prepare(Map<String, Object> topoConf, TopologyContext context, OutputCollector collector) {
+		// Uncomment if extending StatusEmitterBolt
+		//super.prepare(topoConf, context, collector);
+
+		this._collector = collector;
+
+		this.piiRedacter = getInstance(topoConf);
+		LOG.info("Initialized PiiRedacter instance");
+
+		// Get language metadata field name
+		String confLanguageField = ConfUtils.getString(topoConf, "pii.language.field");
+		if (confLanguageField != null && !confLanguageField.isEmpty()) {
+			languageFieldName = confLanguageField;
+		}
+		LOG.info("PII language field: {}", languageFieldName);
+
+		piiEnabled = ConfUtils.getBoolean(topoConf, PII_ENABLE_FIELD, false);
+		LOG.info("PII disabled: {}", piiEnabled);
+
+	}
+
+	@Override
+	public void execute(Tuple input) {
+		String url = input.getStringByField(FIELD_URL);
+		Metadata metadata = (Metadata) input.getValueByField(FIELD_METADATA);
+		String text = input.getStringByField(FIELD_TEXT);
+		byte[] originalBytes = input.getBinaryByField(FIELD_CONTENT);
+
+		LOG.info("Processing URL for PII redaction: {}", url);
+
+		if (!piiEnabled) {
+			emitTuple(input, url, originalBytes, metadata, text);
+			this._collector.ack(input);
+			return;
+		}
+		
+		if (StringUtils.isBlank(text)) {
+			LOG.info("No text to process for URL: {}", url);
+			metadata.addValue("pii.processed", "false");
+			// Force the binary content to a dummy content
+			emitTuple(input, url, REDACTED_BYTES, metadata, "");
+			this._collector.ack(input);
+			return;
+		}
+
+		try {
+			String language = metadata.getFirstValue(languageFieldName);
+			String redacted = (language != null) ?
+					piiRedacter.redact(text, language) :
+					piiRedacter.redact(text);
+
+			if (redacted == null) {
+				throw new Exception("PII Redacter returned null");
+			}
+
+			metadata.addValue("pii.processed", "true");
+			
+			// Force the binary content to a dummy content
+			emitTuple(input, url, REDACTED_BYTES, metadata, redacted);
```

Comment:
Why do we force the binary content to dummy content here, if we have `redacted`? Shouldn't we set it to `redacted.getBytes(...)` ?